cabal-install/src/Distribution/Client/FileMonitor.hs

```diff
@@ -889,20 +893,20 @@ buildMonitorStateFileSet
   -- ^ existing file hashes
   -> FilePath
   -- ^ root directory
-  -> [MonitorFilePath]
+  -> Set MonitorFilePath
   -- ^ patterns of interest
   --   relative to root
   -> IO MonitorStateFileSet
 buildMonitorStateFileSet mstartTime hashcache root =
-  go [] []
+  go mempty mempty . Set.toList
   where
     go
-      :: [MonitorStateFile]
-      -> [MonitorStateGlob]
+      :: Set MonitorStateFile
+      -> Set MonitorStateGlob
       -> [MonitorFilePath]
       -> IO MonitorStateFileSet
     go !singlePaths !globPaths [] =
-      return (MonitorStateFileSet (reverse singlePaths) (reverse globPaths))
+      return (MonitorStateFileSet singlePaths globPaths)
```

Comment:
This worries me: `reverse` implies the ordering is significant. (I know it's reversing because it uses `:` to prepend paths, but that still means the original path order is considered important enough to reconstruct afterward.)

Reply:
Yeah, this worried me a bit too. I figure this was just out of habit, but I might be wrong.